### PR TITLE
Add a comment explaining what we do in the implement_vertex macro

### PR DESCRIPTION
**Pull request by est31**
Also add a forgotten attr_type_of_val invocation.

### CODE REVIEW

File: src/macros.rs [hunk 1]
```diff
@@ -131,12 +131,15 @@ macro_rules! implement_vertex {
...
                             {
+                                // Obtain the type of the $field_name field of $struct_name and
+                                // call get_type on it.
                                 fn attr_type_of_val<T: $crate::vertex::Attribute>(_: Option<&T>)
                                     -> $crate::vertex::AttributeType
                                 {
                                     <T as $crate::vertex::Attribute>::get_type()
                                 }
-                                attr_type_of_val(None::<&$struct_name>.map(|v| v.$field_name))
+                                let field_option = None::<&$struct_name>.map(|v| v.$field_name);
```
Comment (RalfJung) [inline src/macros.rs]: Why isn't this `&v.$field_name`?

Comment (RalfJung) [reply src/macros.rs]: How did this even compile before?^^

Comment (est31) [reply src/macros.rs]: The macro itself compiled but expansion only happens in the tests which right now always fail on travis. So I didn't notice that the test for `implement_vertex!` failed as well.

**Pull request merged**